### PR TITLE
fix(AoTPlugin): don't override context module deps

### LINKED ISSUE #2496: Lazy loading is not working together with 'import *' syntax
> ## Please provide us with the following information:
### OS?

> Mac OSX El Capitan
### Versions.

angular-cli: 1.0.0-beta.16
node: 5.7.1
os: darwin x64
### Repro steps.

> Was this an app that wasn't created using the CLI? What change did you do on your code? etc.
> The app was not created using the CLI. When upgrading to Angular 2 I choose to use the CLI as serving and building tool

I have this route in my app.routing.ts:

``````
const ROUTES: Routes = [
    { path: '', redirectTo: 'analyse', pathMatch: 'full' },
    { path: 'annotate', component: AnnotateComponent },
    { path: 'fraud', component: FraudComponent },
    { path: 'info',  loadChildren: 'app/info/info.module#InfoModule'  },
];```

the InfoModule should be loaded lazily when navigating to it's path. But even in de --prod build there's no seperate bundle. The InfoModule is still part of the main.bundle.js.

Note that I'm still importing the InfoModule in the app.module.ts otherwise the info route does not work at all.
``````

### LINKED ISSUE #3079: Dynamically require module
I need to dynamically require a module.
In webpack I did it via require.ensure, but this is not working.

**Case:** 
I am getting data from the server and have to map these to a ts classes and create an object tree. 
The data can be nested infinitely, so I have to load the module dynamic depending on a type attached to the data.

### PR DESCRIPTION
**Pull request by filipesilva**
Thanks to @clydin for a much cleaner solution!

Fix #2496
Fix #3079

### CODE REVIEW

File: tests/e2e/tests/misc/lazy-module.ts [hunk 1]
```diff
@@ -21,12 +22,42 @@ export default function() {
...
+      if (oldNumberOfFiles >= currentNumberOfDistFiles) {
+        throw new Error('Bundles were not created after adding \'import *\'.');
+      }
     })
     // Check for AoT and lazy routes.
     .then(() => ng('build', '--aot'))
     .then(() => readdirSync('dist').length)
     .then(currentNumberOfDistFiles => {
-      if (oldNumberOfFiles >= currentNumberOfDistFiles) {
+      if (oldNumberOfFiles != currentNumberOfDistFiles) {
```
Comment (hansl) [inline tests/e2e/tests/misc/lazy-module.ts]: The old number of files should be smaller, so I think this change is unneeded (and can lead to errors)?

File: packages/@ngtools/webpack/src/plugin.ts [hunk 2]
```diff
@@ -210,16 +215,24 @@ export class AotPlugin implements Tapable {
...
         }
 
-        this.done.then(() => {
-          result.resource = this.skipCodeGeneration ? this.basePath : this.genDir;
-          result.recursive = true;
-          result.dependencies.forEach((d: any) => d.critical = false);
-          result.resolveDependencies = createResolveDependenciesFromContextMap(
-            (_: any, cb: any) => cb(null, this._lazyRoutes));
-
+        // there is no way to find the original request, so try to
```
Comment (hansl) [inline packages/@ngtools/webpack/src/plugin.ts]: You can use a special property set by the function above, that should work, right?

Comment (filipesilva) [inline packages/@ngtools/webpack/src/plugin.ts]: Nope, tried that. It's a whole new object with just a few props picked. Tried a bunch of stuff even. Also checked with @TheLarkInn, he didn't know of a way.

File: tests/e2e/tests/misc/lazy-module.ts (hunk 1 above)
Comment (filipesilva) [inline tests/e2e/tests/misc/lazy-module.ts]: I now update the number: `oldNumberOfFiles = currentNumberOfDistFiles;`.

File: packages/@ngtools/webpack/src/plugin.ts (hunk 2 above)
Comment (filipesilva) [inline packages/@ngtools/webpack/src/plugin.ts]: @clydins approach does away with this entirely, and makes it possible to always target the right resource.

**Comment by born2net** [comment]
sweet, need to wait or the next NPM build to test this?

regards

**Comment by filipesilva** [comment]
@born2net after it is accepted, yes. You can also use master if you want.

**Comment by born2net** [comment]
tx man appreciate the efforts

**Comment by hansl** [comment]
LGTM, looks much better!

**Pull request merged**